test/e2e/node/security_context.go

```diff
@@ -128,6 +128,48 @@ var _ = SIGDescribe("Security Context [Feature:SecurityContext]", func() {
 		})
 	})
 
+	It("should support container.SecurityContext.RunAsNonRootGroup [Feature:RunAsGroup]", func() {
```

Comment:
I'd like to also see some tests verifying the behavior when different ways of setting the group are used, including:

1. Dockerfile sets the group through the USER directive
2. Root group is added to a non-root user in the `/etc/group` file